vyper/venom/analysis/cfg.py

```diff
@@ -32,7 +32,8 @@ def analyze(self) -> None:
                 in_bb.add_cfg_out(bb)
 
     def invalidate(self):
-        from vyper.venom.analysis import DominatorTreeAnalysis, LivenessAnalysis
+        from vyper.venom.analysis import DFGAnalysis, DominatorTreeAnalysis, LivenessAnalysis
 
+        self.analyses_cache.invalidate_analysis(DFGAnalysis)
```

Comment:
dfg invalidation invalidates liveness analysis

Reply:
actually, why should cfg invalidation invalidate the dfg?

Reply:
It's safer to assume that something that requested the invalidation of the cfg to have caused some changes that invalidate the dfg. I think the problem stems from the original decision to state invalidations instead of "preservations". I have been wanting to change that since day one. It would be better to have passes state what they preserve, and all other things get invalidated.